src/window/window_wrapper.rs

```diff
@@ -30,9 +30,11 @@ const MIN_WINDOW_WIDTH: u64 = 20;
 const MIN_WINDOW_HEIGHT: u64 = 6;
 
 pub struct WinitWindowWrapper {
-    pub windowed_context: WindowedContext,
-    skia_renderer: SkiaRenderer,
+    // Don't rearrange this, unless you have a good reason to do so
+    // The destruction order has to be correct
     renderer: Renderer,
+    skia_renderer: SkiaRenderer,
+    pub windowed_context: WindowedContext,
```

Comment:
Does "has to be correct" express an implicit unsafe invariant to uphold? Or just a logical error if changed without much thought?

Reply:
It hangs if you delete the opengl context before Skia, The rest are not that important, but this is the order I have tested with and verified at least quickly to work.